internal/graph/deletablememory_test.go

```diff
@@ -0,0 +1,412 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package graph
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"reflect"
+	"testing"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/internal/cas"
+	"oras.land/oras-go/v2/internal/descriptor"
+)
+
+/*
+A(manifest)-------------------+
+
+	|                         |
+	|                         |
+	|                         |
+	v                         |
+
+B(manifest)-----------------+ |
+
+	|                       | |
+	|                       | |
+	v                       v v
+
+C(layer)                  D(layer)
+*/
```

Comment:
LOL this looks ok in VSCode, but looks strange on GitHub😂

Reply:
redrew the graphs and refined them.